plugins/woocommerce/includes/admin/helper/class-wc-helper.php

```diff
@@ -681,6 +684,55 @@ public static function current_screen( $screen ) {
 		}
 	}
 
+	private static function maybe_redirect_to_new_marketplace_installer() {
+		// Redirect requires the "install" URL parameter to be passed.
+		if ( empty( $_GET[ 'install' ] ) ) {
+			return;
+		}
+
+		$my_subscriptions_url = self::get_subscriptions_redirect_url( array( 'install' => wp_unslash( $_GET[ 'install' ] ) ), true );
+		wp_safe_redirect( esc_url_raw( $my_subscriptions_url ) );
+	}
+
+	/**
+	 * Get the subscriptions page url.
+	 * This provides compatibility for the old and new subscriptions page and replaces
+	 * the previous `get_helper_redirect_url` method below.
+	 *
+	 * @param array $args Query args added to the base url.
+	 * @return string
+	 */
+	public static function get_subscriptions_redirect_url( array $args = array(), bool $redirect_to_wc_admin = false ): string {
```

Comment:
After some discussion, it felt like it made more sense to create a new method, rather than reuse the `get_helper_redirect_url` method. Modifying the existing one would require additional work and modifying all dependent methods for it to be compatible. This prevents the need to do so and gives us a more flexible redirect method.